sklearn/inspection/_partial_dependence.py

```diff
@@ -416,4 +477,16 @@ def partial_dependence(estimator, X, features, response_method='auto',
     averaged_predictions = averaged_predictions.reshape(
         -1, *[val.shape[0] for val in values])
 
-    return averaged_predictions, values
+    if kind == 'average':
+        warnings.warn(
+            "A Bunch will be returned in place of 'predictions' from 0.25 with"
```

Comment:
This is not appropriate management of backwards compatibility. Firstly, we should be telling the user to pass `kind='both'` in the meantime to get the future behaviour. Secondly, this entails the default behaviour (when `kind` is unspecified) changing at version 0.25.

I'm not entirely sure what the intention is here, but you might want to carefully review https://scikit-learn.org/dev/developers/contributing.html#maintaining-backwards-compatibility to get a sense of how this should be done.

Reply:
Thanks for the link. While it does not cover the exact scenario, it is helpful in getting a sense of how this should be done.
The first concern is addressed in https://github.com/scikit-learn/scikit-learn/pull/16619/commits/2d8b322f8d82964eb069ade4971ea733889c2222. 
I am not sure what to do about the second concern. Isn't this why we give a warning for two versions? Would you like the fact that the default behaviour is changing to be highlighted more prominently in the warning?

Reply:
I suppose what I'm concerned about is you give the user no way to specify kind='average' that would allow them to get the future result of kind='average'. So kind='average' is meaningless until 0.23. In which case you should rather have the default be kind='legacy' to warn that defaults will change, while kind='average' does what it will do in the future.

Reply:
Makes sense. Thanks for the clarification. Done in https://github.com/scikit-learn/scikit-learn/pull/16619/commits/5842f36fd165d22529ab95d88a08b561a6f715ff